audio/src/fetch.rs

```diff
@@ -1,17 +1,21 @@
-use crate::range_set::{Range, RangeSet};
+use std::task::Poll;
+use std::task::Context;
+use std::pin::Pin;
 use byteorder::{BigEndian, ByteOrder, WriteBytesExt};
 use bytes::Bytes;
-use futures::sync::{mpsc, oneshot};
-use futures::Stream;
-use futures::{Async, Future, Poll};
+use futures::channel::{mpsc, oneshot};
+use futures::{ready, Future, Stream};
+use log::{debug, warn, trace};
+use crate::range_set::{Range, RangeSet};
 use std::cmp::{max, min};
 use std::fs;
 use std::io::{self, Read, Seek, SeekFrom, Write};
 use std::sync::{Arc, Condvar, Mutex};
 use std::time::{Duration, Instant};
+use tokio::prelude::*;
 use tempfile::NamedTempFile;
 
-use futures::sync::mpsc::unbounded;
+use futures::channel::mpsc::unbounded;
```

Comment:
Sorry just stumbled on this PR. You should use these exports from `tokio` so you don't have to worry about versioning. Same with `Stream`, `ready`,  etc. `Future` can come from `std::future::Future`